crates/router/src/core/payments/tokenization.rs

```diff
@@ -515,35 +515,79 @@ where
                         }
                     },
                     None => {
-                        let pm_metadata = create_payment_method_metadata(None, connector_token)?;
-
-                        locker_id = resp.payment_method.and_then(|pm| {
-                            if pm == PaymentMethod::Card {
-                                Some(resp.payment_method_id)
-                            } else {
-                                None
+                        let customer_apple_pay_saved_pm_id_option = if payment_method_type
+                            == Some(api_models::enums::PaymentMethodType::ApplePay)
+                        {
+                            match state
+                                .store
+                                .find_payment_method_by_customer_id_merchant_id_list(
```

Comment:
pass the PM type directly in the query and return top 1. return Option<String> to avoid iterating the list.

Reply:
We do not have db interface that accepts payment method type and list the payment method of the passed payment method type for given customer.